src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -527,6 +528,25 @@ private Expression CreateTotalCountExpression(Expression source, ExpandedNavigat
             return PropertyContainer.CreatePropertyContainer(includedProperties);
         }
 
+        private static SelectExpandClause GetOrCreateSelectExpandClause(KeyValuePair<IEdmNavigationProperty, ExpandedReferenceSelectItem> propertyToExpand)
+        {
```

Comment:
What about $expand=NP/$ref (query-options)

Reply:
I assume this just affects the select list, and you would still be able to filter/sort/etc. on non-projected properties, but it would be good to have a test to validate this works.

---
In reply to: [295421753](https://github.com/OData/WebApi/pull/1797#discussion_r295421753) [](ancestors = 295421753)

Reply:
Added three related test cases. 

---
In reply to: [298729805](https://github.com/OData/WebApi/pull/1797#discussion_r298729805) [](ancestors = 298729805,295421753)